test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/MapTest.scala

```diff
@@ -202,29 +323,50 @@ trait MapTest {
     if (factory.allowsNullValuesQueries)
       assertFalse(mp.containsValue(null))
     else
-      expectThrows(classOf[Throwable], mp.containsValue(null))
+      expectThrows(classOf[NullPointerException], mp.containsValue(null))
   }
 
-  @Test def should_give_proper_Collection_over_values(): Unit = {
+  @Test def testPutAll(): Unit = {
     val mp = factory.empty[String, String]
 
     mp.put("ONE", "one")
-    val values = mp.values
+
+    mp.putAll(TrivialImmutableMap("X" -> "y", "A" -> "b"))
+    assertEquals(3, mp.size())
+    assertEquals("one", mp.get("ONE"))
+    assertEquals("y", mp.get("X"))
+    assertEquals("b", mp.get("A"))
+
+    val nullMap = TrivialImmutableMap((null: String) -> "y", "X" -> "z")
+    if (factory.allowsNullKeys) {
+      mp.putAll(nullMap)
+      assertEquals("y", mp.get(null))
+      assertEquals("z", mp.get("X"))
+      assertEquals("one", mp.get("ONE"))
+      assertEquals("b", mp.get("A"))
+    } else {
+      expectThrows(classOf[NullPointerException], mp.putAll(nullMap))
```

Comment:
Is it specified how the map looks after throwing this? If yes, test?

Reply:
No, I could not find any specification for that. All we know is that an NPE will be thrown, but AFAICT the map have stored some or none of the other elements before throwing.